File: fhirengine/src/main/java/com/google/fhirengine/index/ReferenceIndex.java
package com.google.fhirengine.index;

import com.google.auto.value.AutoValue;

/** An reference index for a specific resource. */
@AutoValue
public abstract class ReferenceIndex {
  public static ReferenceIndex create(String name, String path, String value) {
    return new AutoValue_ReferenceIndex(name, path, value);
  }

  /** The name of the string index, e.g. "given". */
  public abstract String name();

  /** The path of the string index, e.g. "Patient.name.given". */
  public abstract String path();

  /** The value of the string index, e.g. "Tom". */
  public abstract String value();
}

File: engine/src/main/java/com/google/android/fhir/index/entities/ReferenceIndex.kt
/*
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.google.android.fhir.index.entities

/**
 * An index record for a reference value in a resource.
 *
 * See https://hl7.org/FHIR/search.html#reference.
 */
internal data class ReferenceIndex(
  /** The name of the reference index, e.g. "subject". */
  val name: String,
  /** The path of the reference index, e.g. "Observation.subject". */
  val path: String,
  /** The value of the reference index, e.g. "Patient/123". */
  val value: String
)
